### PR TITLE
test: assert external tsconfig is usable with workers

### LINKED ISSUE #140: ts_config extends from external package
I'm trying to extend a tsconfig from an external dependency like this:
```
ts_config(
    name = "tsconfig",
    src = "tsconfig.json",
    deps = ["//:node_modules/@tsconfig/node16-strictest-esm"],
)
```

```
{
  "extends": "@tsconfig/node16-strictest-esm/tsconfig.json"
}
```

I have also tried adding `/dir` to the package, and still not working.

```
ERROR: /home/juan/Projects/hibot/rocka/proto-generator/BUILD:10:11: Compiling TypeScript project //proto-generator:ts [tsc -p proto-generator/tsconfig.json] failed: (Exit 1): tsc_worker.sh failed: error executing command bazel-out/k8-opt-exec-2B5CBBC6/bin/external/npm_typescript/tsc_worker.sh @bazel-out/k8-fastbuild/bin/proto-generator/src/index.js-0.params
Compilation failed:	
 TypeError: Cannot use 'in' operator to search for 'tsconfig.json' in undefined
    at getNode (/home/juan/.cache/bazel/_bazel_juan/110af2bd36a97b83ddd5caaaec0d3270/execroot/rocka/bazel-out/k8-opt-exec-2B5CBBC6/bin/external/npm_typescript/tsc_worker.sh.runfiles/npm_typescript/ts_project_worker.js:57:24)
```

### PR DESCRIPTION
**Pull request by thesayyn**
fixes #140

### CODE REVIEW

File: examples/pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Comment (jbedard) [inline examples/pnpm-lock.yaml:1]: This is being downgraded? Is your local pnpm old?

File: examples/pnpm-lock.yaml [hunk 2]
```diff
@@ -17,13 +18,14 @@ specifiers:
 
 dependencies:
   '@angular/compiler': 14.2.8_@angular+core@14.2.8
-  '@angular/compiler-cli': 14.2.8_4i3i6fp4ncuzdz7mihas2rstji
+  '@angular/compiler-cli': 14.2.8_e2368f15fc68a991e7ec41c12d46534a
```
Comment (jbedard) [inline examples/pnpm-lock.yaml:21]: Why is angular being updated?

File: examples/pnpm-lock.yaml (hunk 1 above)
Comment (thesayyn) [reply examples/pnpm-lock.yaml:1]: probably. I'll fix it.

File: examples/tsconfig_external/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+ts_config(
+    name = "config",
+    src = "tsconfig.json",
+    deps = [
+        "//examples:node_modules/@tsconfig/node16-strictest-esm"
+    ],
+)
+
+ts_project(
```
Comment (jbedard) [inline examples/tsconfig_external/BUILD.bazel:11]: Do we need a target to actually run this as a test?

Comment (thesayyn) [reply examples/tsconfig_external/BUILD.bazel:11]: yes. we need to assert that the ts_project worker can read the external tsconfig properly. ts_config target by itself is not enough to assert that.

File: examples/pnpm-lock.yaml [hunk 4]
```diff
@@ -9,21 +9,23 @@ specifiers:
...
   '@babel/types': 7.18.6
+  '@tsconfig/node16-strictest-esm': ^1.0.3
   '@types/jasmine': '*'
   '@types/node': 18.6.1
   date-fns: 2.29.1
   rxjs: '7'
   typescript: 4.8.2
 
 dependencies:
-  '@angular/compiler': 14.2.8_@angular+core@14.2.8
```
Comment (jbedard) [inline examples/pnpm-lock.yaml:19]: This is still changing far more then necessary. If you revert and just run `pnpm install @tsconfig/node16-strictest-esm` does it still have this large diff? I'd think it should be 1 line...

File: examples/pnpm-lock.yaml (hunk 2 above)
Comment (thesayyn) [reply examples/pnpm-lock.yaml]: probably because I used an old version of pnpm to generate the lock file.

File: examples/pnpm-lock.yaml (hunk 4 above)
Comment (thesayyn) [reply examples/pnpm-lock.yaml]: already fixed.

File: examples/tsconfig_external/BUILD.bazel (hunk 3 above)
Comment (jbedard) [reply examples/tsconfig_external/BUILD.bazel:11]: I mean adding a `build_test` or something to create an actual test target...

Comment (thesayyn) [reply examples/tsconfig_external/BUILD.bazel:11]: it's not needed.

Comment (jbedard) [reply examples/tsconfig_external/BUILD.bazel:11]: A lot of them do have test targets and I think we should move that direction. Why not?

Comment (thesayyn) [reply examples/tsconfig_external/BUILD.bazel:11]: examples meant to be for users to understand how to do `x` and none of them has a `build_test` target. If we are going to start putting build_test to all of the examples that should be filed and fixed separately. I just follow the convention that we have today.

Comment (jbedard) [reply examples/tsconfig_external/BUILD.bazel:11]: About 2/3 seem to have tests already, mainly `build_test` and `assert_contains`.

I do think they should all have tests though, probably either `build_test`, `assert_contains` or `write_source_file`. Want to do that in a followup? Or maybe I will 🤷

**Review by jbedard — APPROVED** [review]

**Pull request merged**